src/object/cli_mod.c

```diff
@@ -19,132 +19,43 @@
 unsigned int	srv_io_mode = DIM_DTX_FULL_ENABLED;
 int		dc_obj_proto_version;
 
-struct obj_proto {
-	struct rsvc_client	cli;
-	crt_endpoint_t		ep;
-	int			version;
-	int			rc;
-	bool			completed;
-};
-
-static void
-query_cb(struct crt_proto_query_cb_info *cb_info)
-{
-	struct obj_proto *oproto = (struct obj_proto *)cb_info->pq_arg;
-
-	if (daos_rpc_retryable_rc(cb_info->pq_rc)) {
-		uint32_t	ver_array[2] = {DAOS_OBJ_VERSION - 1, DAOS_OBJ_VERSION};
-		int		rc;
-
-		rc = rsvc_client_choose(&oproto->cli, &oproto->ep);
-		if (rc) {
-			D_ERROR("rsvc_client_choose() failed: "DF_RC"\n", DP_RC(rc));
-			oproto->rc = rc;
-			oproto->completed = true;
-		}
-
-		rc = crt_proto_query_with_ctx(&oproto->ep, obj_proto_fmt_0.cpf_base, ver_array, 2,
-					      query_cb, oproto, daos_get_crt_ctx());
-		if (rc) {
-			D_ERROR("crt_proto_query_with_ctx() failed: "DF_RC"\n", DP_RC(rc));
-			oproto->rc = rc;
-			oproto->completed = true;
-		}
-	} else {
-		oproto->rc = cb_info->pq_rc;
-		oproto->version = cb_info->pq_ver;
-		oproto->completed = true;
-	}
-}
-
 /**
  * Initialize object interface
  */
 int
 dc_obj_init(void)
 {
 	uint32_t		ver_array[2] = {DAOS_OBJ_VERSION - 1, DAOS_OBJ_VERSION};
-	struct dc_mgmt_sys	*sys;
-	struct obj_proto	*oproto = NULL;
-	crt_context_t		ctx = daos_get_crt_ctx();
-	int			num_ranks;
 	int			rc;
 
 	rc = obj_utils_init();
 	if (rc)
-		D_GOTO(out, rc);
+		return rc;
 
 	rc = obj_class_init();
 	if (rc)
 		D_GOTO(out_utils, rc);
 
 	dc_obj_proto_version = 0;
-
-	rc = dc_mgmt_sys_attach(NULL, &sys);
-	if (rc != 0) {
-		D_ERROR("failed to attach to grp rc "DF_RC"\n", DP_RC(rc));
+	rc = daos_rpc_proto_query(obj_proto_fmt_0.cpf_base, ver_array, 2, &dc_obj_proto_version);
```

Comment:
I would not suggest changes in this patch, but maybe in the future object, pool, and container can all have some convention for naming these crt_proto_format structures/variable names to represent whatever the current and current-1 versions are without including the actual version number in the variable name. object is using fmt_1 and fmt_0 to represent (DAOS_OBJ_VERSION and DAOS_OBJ_VERSION-1 that are currently 9 and 8 respectively). So this is a pretty good model. But pool and container are using structure names like fmt_5 and fmt_4, or fmt_7 and fmt_6, with the actual protocol version number in the structure/variable name.

Reply:
So do you think fmt_1 and fmt_0 is good practice? i remembered pool/containers fmt_5 xx followed @liw suggestions. i am good with either way, but yup, it is better to have same convention.

Reply:
Yeah, would need to think about it in a little more detail to be sure. fmt_1/0 or perhaps fmt_cur/prev might be reasonable.